protocol/git/README.md

```diff
@@ -1,120 +1,50 @@
-
-    git push origin --delete <branch-name>
-
-Delete your local feature branch.
-
-    git branch --delete <branch-name>
-
-[find the pull request]: http://stackoverflow.com/a/17819027
+# Proposal: Use release branches and cease automatic staging
+
+Our current process for integrating, delivering, and deploying to healthify:
+
+1. Create a changeset on a development branch forked from `master`.
+2. Submit a pull request to merge the branch into `master`.
+3. Review the changes in the pull request.
+4. Merge the development branch into `master`, triggering delivery to staging.
+5. Conduct manual QA through the front end of the staging app.
+
+This process is conducted _concurrently_ for multiple changesets. If we have at least one changeset that has passed all 5 steps, and none that are between steps 4 and 5, we are ready to deploy.
+
+We recognize three resulting problems:
+
+1. We cannot deploy when there is any changeset that has finished 1-4 but not 5.
+2. Sometimes, features are constituted by many distinct changesets.
+3. Sometimes, a set of many features should be deployed simultaneously.
+
+Therefore, we propose a two-part solution: Use release branches to integrate multiple dependent changesets, and decouple the post-code-review integration from the staging deployment.
+
+1. Use release branches. Every branch to be merged into master should represent an atomically deployable feature set (i.e,. a release). Changes should be committed on _branches off of the release branch_ as opposed to branches off of master. There should be at least one development branch for each release branch; there can be multiple development branches for a release branch, to map better to Pivotal stories or to yield more digestible PRs.
+2. Cease automatic staging deployment. As each component development branch is reviewed, revised, and approved it should be _merged into the release branch_. Therefore, as development progresses, the release branch will accumulate merges of approved development branches. Given a release branch, `my-release`, whose component branches have all passed review, QA may be conducted when ready, according to this console-driven process:
+  1. Pull `origin/master` to the local `master`.
+  2. Fork a new local `stage-my-release` branch off of `master`.
+  3. Merge `origin/my-release` into `stage-my-release`.
+  4. Push `stage-my-release` to `staging/master`.
+  5. Wait for the staging deploy to finish.
+  6. Conduct QA through the staging front-end.
+  7. If QA is successful:
+    1. Merge `stage-my-release` back into `master`.
+    2. Push `master` to `origin/master`.
+    3. Push `master` to `staging/master`.
+    4. Push `master` to `production/master`, deploying it to production. (As a result, we no longer need an `origin/production` branch.)
+  8. If QA is not successful:
+    1. Indicate that the feature-set was rejected, in Pivotal Tracker.
+    2. Push `origin/master` to `staging/master` to reset the state of staging.
```

Comment:
> We should also ensure that we rollback DB migrations!

+1 ...but what if we reset the staging DB (create from scratch and load some dump of "fixtures") on every staging deploy, the same way we set up for tests...?